Mathlib/Data/Option/Basic.lean

```diff
@@ -105,6 +105,10 @@ theorem bind_eq_some' {x : Option α} {f : α → Option β} {b : β} :
 
 #align option.bind_eq_none' Option.bind_eq_none'
 
+theorem bind_congr {f g : α → Option β} {x : Option α}
```

Comment:
I think this should be tagged `@[congr]`, if Lean allows that?

Reply:
I think not. If we were to create a congr theorem, the following theorem would be more appropriate.:
```lean
@[congr]
theorem bind_congr' {f g : α → Option β} {x y : Option α} (hx : x = y)
    (hf : ∀ a ∈ y, f a = g a) : x.bind f = y.bind g :=
  hx.symm ▸ bind_congr hf
```

Reply:
`congr` theorem example: [`dite_congr`](https://leanprover-community.github.io/mathlib4_docs/Init/SimpLemmas.html#dite_congr)

Reply:
I've added this new theorem.